### PR TITLE
docs(disp): add rotation examples

### PR DESCRIPTION
**Pull request by kisvegabor**
A clear and concise description of what the bug or new feature is.

### Notes
- Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed.
- Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant.
- Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/master/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/master/Kconfig).
- Run `scripts/code-format.py` (`astyle v3.4.12` needs to installed by running `cd scripts; ./install_astyle.sh`) and follow the [Code Conventions](https://docs.lvgl.io/master/CODING_STYLE.html).
- Mark the Pull request as [Draft](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/changing-the-stage-of-a-pull-request) while you are working on the first version, and mark is as _Ready_ when it's ready for review.
- When changes were requested, [re-request review](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review) to notify the maintainers.
- Help us to review this Pull Request! Anyone can [approve or request changes](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/approving-a-pull-request-with-required-reviews).

### CODE REVIEW

File: docs/details/main-components/display.rst [hunk 1]
```diff
@@ -403,6 +404,82 @@ The same is true for :cpp:enumerator:`LV_DISPLAY_RENDER_MODE_FULL`.
 In the case of :cpp:enumerator:`LV_DISPLAY_RENDER_MODE_PARTIAL` the small rendered areas
 can be rotated on their own before flushing to the frame buffer.
 
+Below is an example for rotatig when the renderig mode is
```
Comment (XuNeo) [inline docs/details/main-components/display.rst]:
```suggestion
Below is an example for rotating when the rendering mode is
```

File: docs/details/main-components/display.rst [hunk 2]
```diff
@@ -403,6 +404,82 @@ The same is true for :cpp:enumerator:`LV_DISPLAY_RENDER_MODE_FULL`.
...
+			/*Use the rotated area and rotated buffer from now on*/
+			area = &rotated_area;
+			px_map = rotated_buf;
+		}
+
+	    my_set_window(area->x1, area->y1, area->x2, area->y2);
+	    my_send_colors(px_map);
+	}
+
+Below is an example for rotatig when the renderig mode is
```
Comment (XuNeo) [inline docs/details/main-components/display.rst]:
```suggestion
Below is an example for rotating when the rendering mode is
```

**Review by XuNeo — DISMISSED** [review]

File: docs/details/main-components/display.rst [hunk 3]
```diff
@@ -403,6 +404,82 @@ The same is true for :cpp:enumerator:`LV_DISPLAY_RENDER_MODE_FULL`.
 In the case of :cpp:enumerator:`LV_DISPLAY_RENDER_MODE_PARTIAL` the small rendered areas
 can be rotated on their own before flushing to the frame buffer.
 
+Below is an example for rotating when the rendering mode is
+:cpp:enumerator:`LV_DISPLAY_RENDER_MODE_PARTIAL` and the rotated image should be sent to a
+**display controller**.
+
+.. code-block:: c
+	/*Rotate a partially rendered area to another buffer and send it*/
+	void flush_cb(lv_display_t * disp, const lv_area_t * area, uint8_t * px_map
```
Comment (liamHowatt) [inline docs/details/main-components/display.rst]:
```suggestion
	void flush_cb(lv_display_t * disp, const lv_area_t * area, uint8_t * px_map)
```

File: docs/details/main-components/display.rst [hunk 4]
```diff
@@ -403,6 +404,82 @@ The same is true for :cpp:enumerator:`LV_DISPLAY_RENDER_MODE_FULL`.
...
+			lv_draw_sw_rotate(px_map, rotated_buf, src_w, src_h, src_stride, dest_stride, rotation, cf);
+
+			/*Use the rotated area and rotated buffer from now on*/
+			area = &rotated_area;
+			px_map = rotated_buf;
+		}
+
+	    my_set_window(area->x1, area->y1, area->x2, area->y2);
+	    my_send_colors(px_map);
+	}
```
Comment (liamHowatt) [inline docs/details/main-components/display.rst]: Indentation was mixed spaces and tabs. Converted to all spaces.

```suggestion
    /*Rotate a partially rendered area to another buffer and send it*/
    void flush_cb(lv_display_t * disp, const lv_area_t * area, uint8_t * px_map)
    {
        lv_display_rotation_t rotation = lv_display_get_rotation(disp);
        lv_area_t rotated_area;
        if(rotation != LV_DISPLAY_ROTATION_0) {
            lv_color_format_t cf = lv_display_get_color_format(disp);
            /*Calculate the position of the rotated area*/
            rotated_area = *area;
            lv_display_rotate_area(disp, &rotated_area);
            /*Calculate the source stride (bytes in a line) from the width of the area*/
            uint32_t src_stride = lv_draw_buf_width_to_stride(lv_area_get_width(area), cf);
            /*Calculate the stride of the destination (rotated) area too*/
            uint32_t dest_stride = lv_draw_buf_width_to_stride(lv_area_get_width(&rotated_area), cf);
            /*Have a buffer to store the rotated area and perform the rotation*/
            static uint8_t rotated_buf[500*1014];
            int32_t src_w = lv_area_get_width(area);
            int32_t src_h = lv_area_get_height(area);
            lv_draw_sw_rotate(px_map, rotated_buf, src_w, src_h, src_stride, dest_stride, rotation, cf);
            /*Use the rotated area and rotated buffer from now on*/
            area = &rotated_area;
            px_map = rotated_buf;
        }
        my_set_window(area->x1, area->y1, area->x2, area->y2);
        my_send_colors(px_map);
    }
```

File: docs/details/main-components/display.rst [hunk 5]
```diff
@@ -403,6 +404,82 @@ The same is true for :cpp:enumerator:`LV_DISPLAY_RENDER_MODE_FULL`.
...
+	        for(y = area->y1; y <= area->y2; y++) {
+	            lv_memcpy(fb_start, px_map, src_stride);
+	            px_map += src_stride;
+	            fb_start += fb_stride;
+	        }
+	    }
+	    else {
+	        lv_draw_sw_rotate(px_map, fb_start, src_w, src_h, src_stride, fb_stride, rotation, cf);
+	    }
+	}
```
Comment (liamHowatt) [inline docs/details/main-components/display.rst]:
```suggestion
    /*Rotate a partially rendered area to the frame buffer*/
    void flush_cb(lv_display_t * disp, const lv_area_t * area, uint8_t * px_map)
    {
        lv_color_format_t cf = lv_display_get_color_format(disp);
        uint32_t px_size = lv_color_format_get_size(cf);
        /*Calculate the position of the rotated area*/
        lv_area_t rotated_area = *area;
        lv_display_rotate_area(disp, &rotated_area);
        /*Calculate the properties of the source buffer*/
        int32_t src_w = lv_area_get_width(area);
        int32_t src_h = lv_area_get_height(area);
        uint32_t src_stride = lv_draw_buf_width_to_stride(src_w, cf);
        /*Calculate the properties of the frame buffer*/
        int32_t fb_stride = lv_draw_buf_width_to_stride(disp->hor_res, cf);
        uint8_t * fb_start = my_fb_address;
        fb_start += rotated_area.y1 * fb_stride + rotated_area.x1 * px_size;
        lv_display_rotation_t rotation = lv_display_get_rotation(disp);
        if(rotation == LV_DISPLAY_ROTATION_0) {
            int32_t y;
            for(y = area->y1; y <= area->y2; y++) {
                lv_memcpy(fb_start, px_map, src_stride);
                px_map += src_stride;
                fb_start += fb_stride;
            }
        }
        else {
            lv_draw_sw_rotate(px_map, fb_start, src_w, src_h, src_stride, fb_stride, rotation, cf);
        }
    }
```

File: docs/details/main-components/display.rst [hunk 6]
```diff
@@ -403,6 +404,72 @@ The same is true for :cpp:enumerator:`LV_DISPLAY_RENDER_MODE_FULL`.
...
+            area = &rotated_area;
+            px_map = rotated_buf;
+        }
+        my_set_window(area->x1, area->y1, area->x2, area->y2);
+        my_send_colors(px_map);
+    }
+
+Below is an example for rotating when the rendering mode is
+:cpp:enumerator:`LV_DISPLAY_RENDER_MODE_PARTIAL` and the image can be rotated directly
+into a **frame buffer of the LCD periphery**.
```
Comment (liamHowatt) [inline docs/details/main-components/display.rst]:
```suggestion
into a **frame buffer of the LCD peripheral**.
```

**Review by liamHowatt — APPROVED** [review]

**Review by uLipe — APPROVED** [review]

**Pull request merged**